pysgrid/custom_exceptions.py

```diff
@@ -27,6 +27,18 @@ def __str__(self):
         return error_message
     
     
+class TopologyDimensionError(Exception):
```

Comment:
I am against defining custom Exceptions.  They add a complexity layer to the code that not necessary brings clarity to the reader/user.   In this case, for example, `lvc_split` can be 2D or 3D, if not, it is just a ValueError with the `base_message` you defined.

I see that you have already defined several custom Exceptions, so I will leave to your judgment to keep or remove them.